docs/understand_the_basics.md

```diff
@@ -14,3 +14,74 @@ Before effectively using Argo CD, it is necessary to understand the underlying t
 * If you're integrating with a CI tool:
 	* [GitHub Actions Documentation](https://docs.github.com/en/actions)
 	* [Jenkins User Guide](https://www.jenkins.io/doc/book/)
+
+# Argo CD Contribution Guide
+
+## Explore the Project Structure
+
+To get started with contributing, it's helpful to familiarize yourself with the project structure. Some key folders/files to check:
+- **`cmd/`**: Contains the main entry points for the various Argo CD commands/tools.
+- **`pkg/`**: Contains reusable packages used throughout the project.
+- **`manifests/`**: Kubernetes manifests for deploying Argo CD.
+- **`docs/`**: Documentation files explaining various components.
+- **`README.md`**: Overview of the project, including setup and usage instructions.
+
+---
+
+## Run the Project Locally
+
+To run Argo CD locally, please refer to the official developer guide:
+[Running Locally](https://argo-cd.readthedocs.io/en/stable/developer-guide/running-locally/)
+
+This guide provides comprehensive instructions for setting up and running Argo CD in a local environment. It ensures you follow the recommended practices and avoid common pitfalls.
+
+---
+
+## Start Contributing
+
+Once you're comfortable navigating the codebase, look for issues tagged as **"good first issue"** in the [Argo CD GitHub Issues](https://github.com/argoproj/argo-cd/issues). These are beginner-friendly tasks to help you get started.
+
+Good first issues may be hard to find in a large project like Argo CD with many contributors. If you're unable to find one, follow these steps:
+
+1. **Pick a Component**:
+   - Choose an Argo CD component to focus on (e.g., `repo-server`, `controller`, etc.).
+
+2. **Explore the Codebase**:
+   - Dive into the code for that component.
+   - Look for opportunities to refactor or add tests.
+
+3. **Sort Issues by Label**:
+   - Use the component’s label to filter issues in the repository.
+   - Pick a relevant issue that matches your interests and skills.
```

Comment:
There's a lot to add here and needs to be refactored properly. For example, mention about reading the [component architecture](https://argo-cd.readthedocs.io/en/stable/developer-guide/architecture/components/) and many others.